src/providers/node.rs

```diff
@@ -36,26 +36,47 @@ impl Provider for NpmProvider {
 
     fn setup(&self, app: &App, env: &Environment) -> Result<Option<SetupPhase>> {
         let package_json: PackageJson = app.read_json("package.json")?;
-        let node_pkg = NpmProvider::get_nix_node_pkg(&package_json, env)?;
+        let node_pkg = NodeProvider::get_nix_node_pkg(&package_json, env)?;
+        if NodeProvider::get_package_manager(app)? == "yarn" {
+            let mut yarn_pkg = Pkg::new("yarn");
+            // Only override the node package if not the default one
+            if node_pkg.name != *DEFAULT_NODE_PKG_NAME {
+                yarn_pkg = yarn_pkg.set_override("nodejs", node_pkg.name.as_str());
+            }
+            return Ok(Some(SetupPhase::new(vec![node_pkg, yarn_pkg])));
+        }
 
         Ok(Some(SetupPhase::new(vec![node_pkg])))
     }
 
     fn install(&self, _app: &App, _env: &Environment) -> Result<Option<InstallPhase>> {
```

Comment:
Nit: change `_app` to `app`. Prefixed underscore should only be used if variable is unused

```suggestion
    fn install(&self, app: &App, _env: &Environment) -> Result<Option<InstallPhase>> {
```

Reply:
change `_app` to `app`